### ISSUE TITLE
Coq_nvim as the autocompletion tool

### ISSUE THREAD

**Issue by SpyrosMourelatos** [Feel-ix-343/markdown-oxide#75]
TY for the amazing job I would like to ask if it is possible to use the auto completion features without hrsh7th/nvim-cmp and with ms-jpq/coq_nvim. I have done everything except adding the source
```
{
name = 'nvim_lsp',
  option = {
    markdown_oxide = {
      keyword_pattern = [[\(\k\| \|\/\|#\)\+]]
    }
  }
},
```
<img width="830" alt="Screenshot 2024-04-29 at 11 13 21" src="https://github.com/Feel-ix-343/markdown-oxide/assets/26333139/6df14a30-c007-480c-9e82-233ee7b312ba">

In my configs as I don't use nvim-cmp and I am pretty close to a fully working functionality( see the screenshot)
So I was wondering if we can communicate to ms-joq and add markdown-oxide as a third party source.

 In the next days I will try to manually add it and provide some configs. The next best thing  other than supporting it out of the box is to document the required work to support it.

**Comment by Feel-ix-343** [comment]
Hey thank you so much!! 

I'd love to work with you on this. I have certainly heard of coq but never tried it. 

Let me know how you are developing and I will work on this at some point.

You can check what is being worked on in the linked github project

**Comment by Feel-ix-343** [comment]
If coq_nvim doesn't stop giving completions when you press space (or another weird character), keyword_pattern should not be necessary. 

In my opinion, the concept of keyword_pattern in nvim-cmp is slightly odd. It specifies the pattern that an ongoing completion search text must match - otherwise completions are dropped. This is fine for most LSs because they do not give completion with spaces (which is not in the keyword_pattern by default), but many do. VSCode has nothing like this; there was almost no configuration necessary to get moxide working on it. Keyword pattern is a little odd. 

And you are right about the trigger characters.

**Comment by Feel-ix-343** [comment]
NOTE that keyword_pattern is nvim-cmp specific and is not tied into the language server protocol

**Comment by SpyrosMourelatos** [comment]
Let's close the issue, I think I had something misconfigured on my enormous configs(also maybe the problem happened because my vault is aLlogseq vault). Today I cleaned up my computer and coq_nvim also works. Furthermore, I tested a bit working with ` vim.bo.omnifunc = "v:lua.vim.lsp.omnifunc"` and omnifunc works as well! Here is a screenshot after all :
<img width="1701" alt="Screenshot 2024-04-30 at 19 49 59" src="https://github.com/Feel-ix-343/markdown-oxide/assets/26333139/1f04a202-c7c6-45d6-b758-00e750242dc6">
